src/lib.rs

```diff
@@ -37,6 +37,22 @@ module! {
         pub struct Call as "eth_call"
             (TransactionCall, BlockId) => Vec<u8> [serialization::bytes];
 
+        /// Returns the chain ID of the current network
+        pub struct ChainId as "eth_chainId"
+            Empty => U256;
+
+        /// Generates and returns an estimate of how much gas is necessary to allow the transaction to complete.
+        pub struct EstimateGas as "eth_estimateGas"
+            (TransactionCall, BlockSpec) => U256;
```

Comment:
This inconsistency is kind of funny - `eth_call` allows block hashes, but `eth_estimateGas` does not 😛. I wonder if it is an issue with the spec, or just an oversight...

Reply:
I would have actually tried it out, but I didn't have a transaction in mind...

Reply:
I tried it, and it turns out that nodes accept this. I created an issue in the execution APIs repo to see if it is a spec issue or not.